server/src/main/java/org/elasticsearch/cluster/coordination/stateless/AtomicRegisterPreVoteCollector.java

```diff
@@ -27,11 +31,11 @@ public AtomicRegisterPreVoteCollector(StoreHeartbeatService heartbeatService, Ru
     @Override
     public Releasable start(ClusterState clusterState, Iterable<DiscoveryNode> broadcastNodes) {
         final var shouldRun = new AtomicBoolean(true);
-        heartbeatService.runIfNoRecentLeader(() -> {
+        heartbeatService.checkLeaderHeartbeatAndRun(() -> {
             if (shouldRun.getAndSet(false)) {
                 startElection.run();
             }
-        });
+        }, heartbeat -> logger.info("skip election since there is a recent heartbeat({}) from the leader", heartbeat));
```

Comment:
It felt wrong to add this log in `StoreHeartbeatService` even though it would have worked since aside from tests, it is only used from here.